src/sharepoint/rest/items.ts

```diff
@@ -1,42 +1,186 @@
 "use strict";
+
 import { Queryable } from "./Queryable";
 import * as Util from "../../utils/util";
 import * as Mixins from "./mixins";
+import { Gettable, SelectableGettable, FilterableSelectableGettable } from "./actionables";
+import { Folder } from "./folders";
+import { ContentType } from "./contenttypes";
+import { RoleAssignments } from "./roleassignments";
 
+/**
+ * Describes a collection of Item objects
+ * 
+ */
 export class Items extends Queryable implements Mixins.Gettable, Mixins.Selectable, Mixins.Filterable {
 
-    constructor(url: string | Queryable) {
-        super(url, "items");
+    /**
+     * Creates a new instance of the Items class
+     * 
+     * @param baseUrl The url or Queryable which forms the parent of this fields collection
+     */
+    constructor(baseUrl: string | Queryable) {
+        super(baseUrl, "items");
     }
 
+    /**
+     * Gets an Item by id
+     * 
+     * @param id The integer id of the item to retrieve
+     */
     public getById(id: number): Item {
         this.concat(`(${id})`);
```

Comment:
Read my comment now and didn't really understand it either... See the comment about creating class/interface for Guid above.

Creating types is cheap, I think that the more types we have the better.

What about a getByPartialMatch(entity: Item) which will use whatever is available? (ie name, id or other properties)